triples-generator-api/src/main/scala/io/renku/triplesgenerator/api/events/DatasetViewedEvent.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.triplesgenerator.api.events
+
+import cats.Show
+import cats.syntax.all._
+import io.circe.{Decoder, DecodingFailure, Encoder}
+import io.circe.literal._
+import io.circe.DecodingFailure.Reason.CustomReason
+import io.renku.events.CategoryName
+import io.renku.graph.model.datasets
+
+import java.time.Instant
+
+final case class DatasetViewedEvent(identifier: datasets.Identifier, dateViewed: datasets.DateViewed)
+
+object DatasetViewedEvent {
+
+  def forDataset(identifier: datasets.Identifier, now: () => Instant = () => Instant.now): DatasetViewedEvent =
```

Comment:
cats effect also has this `Clock` typeclass that provides the current time; a bit less convenient but more "pure" 😄

Reply:
That's true. I somehow forgot about it. Maybe next time :)